src/wire.rs

```diff
@@ -200,7 +204,9 @@ impl<S: NetSession> NetTransport<S> {
     fn upgrade(mut session: S, inbound: bool) -> io::Result<Self> {
         session.set_read_timeout(Some(READ_TIMEOUT))?;
         session.set_write_timeout(Some(WRITE_TIMEOUT))?;
-        session.set_nonblocking(true)?;
+        session.set_nonblocking(true)?; // cloudhead: This is already set in `S::connect`,
+                                        // perhaps read/write timeout should be set there as well,
+                                        // then there's no need to "upgrade".
```

Comment:
This part is already addressed and refactored with differentiating blocking and non-blocking connections